### PR TITLE
Add browser.secureStorage proposal

### PR DESCRIPTION
**Pull request by oliverdunk**
This adds the proposal from #154 to a new proposals/ folder in the repository. As discussed at the end of the last meeting, my hope is that this makes it easier to collaborate on the proposal while also providing us with a commit history and issue tracker that we can use while moving this forward as a community.

For now, this is very close to a direct port from the shared document. I plan to start making more changes in future PRs.

### CODE REVIEW

File: proposals/secure-storage.md [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+We’ve historically chosen not to do this because it makes the secrets too easy to access. While a compromised machine is hard to protect against, data in these APIs can be trivially accessed with a console command and no authentication. This allows an “evil maid” attack if a computer is left unlocked, or for another application to read the secret from disk independently of Chrome. While the secret must be in memory while the extension is in use, extensions like password managers aggressively enter a “locked state” after inactivity where secrets are cleared from memory. This means that after only a short period of idle time such an attack would require biometric authentication.
+
+## Remaining Work
+
+- We need to gather feedback from the community and browser vendors to decide which proposal to move forward with, and if any changes are needed.
+- It would be beneficial to support more fine-grained control of the keys generated in Proposal 1. We should document this in more detail, perhaps drawing inspiration from the [Web Crypto API](https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/generateKey).
+
+## Contacts
+
+Oliver Dunk (oliver@1password.com)
```
Comment (Rob--W) [inline proposals/secure-storage.md]: Let's prepend a section "Champions" before the Motivation section, wher the list of people driving the proposal is shown. And include the Github handle, to make it easier for readers/contributors to tag the right person on Github.

Comment (oliverdunk) [reply proposals/secure-storage.md]: I like that! I went ahead and tweaked things as suggested.

File: proposals/secure-storage.md [hunk 2]
```diff
@@ -0,0 +1,158 @@
+# browser.secureStorage
+
+WebExtension proposal to allow the secure storage of data in platform specific locations.
+
+## Champions
+
+- [@oliverdunk](https://github.com/oliverdunk) (1Password) - oliver@1password.com
```
Comment (dotproto) [inline proposals/secure-storage.md]:
```suggestion
- Oliver Dunk (1Password) - [@oliverdunk](https://github.com/oliverdunk), [email](oliver@1password.com)
```

The downside of the change made in response to @Rob--W's [comment](https://github.com/w3c/webextensions/pull/167#discussion_r812960247) is that we lost your name. Perhaps we could tweak the formatting a bit to keep your name and include your GitHub username. WDYT?

**Review by dotproto — APPROVED** [review]
Bikeshedding in champion name formatting aside, I think this is in a good state to merge & iterate on.

File: proposals/secure-storage.md (hunk 2 above)
Comment (oliverdunk) [reply proposals/secure-storage.md]: I'm happy with that! I accepted your suggestion and tweaked the email, let me know what you think. I can't merge this but I think it's good to go, unless you want anyone else to review.

**Review by mukul-p — APPROVED** [review]

File: proposals/secure-storage.md [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+
+**How large could the stored data be?**
+
+The expectation is that implementation specific limits would be set. Only a very small allowance (< 1KB) is needed for the known use cases.
+
+**Are there any extensions that would use this?**
+
+This proposal is drafted by the team at 1Password. Our [extension](https://chrome.google.com/webstore/detail/1password-%E2%80%93-password-mana/aeblfdkhhhdcdjpifhhbdiojplfjncoa?hl=en) is available for Chrome, Firefox, Safari and Edge and has more than 2M weekly active users in the Chrome Web Store alone. We would be quick to implement this in conjunction with or in place of our current Native Messaging solution.
+
+The Dashlane browser extension, also a password manager with over 4M weekly active users in the Chrome Web Store, currently provides a remember me and biometric authentication feature in its extension to make repeated authentication easier. They would also be very quick to implement this in place of their solution which requires a network connection, as they don’t leverage a native application to interface with native storage.
```
Comment (zombie) [inline proposals/secure-storage.md:121]: Do we have a public comment of someone from Dashlane supporting this proposal?  If not, perhaps better to leave this for a later PR they could submit.

**Review by zombie — APPROVED** [review]

File: proposals/secure-storage.md [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+
+Users understandably prefer not to type this, so in desktop applications, a common pattern is to store the decryption key in a system store (like macOS keychain). The system can be told to [only reveal the secret after biometrics have been provided](https://developer.apple.com/documentation/security/keychain_services/keychain_items/restricting_keychain_item_accessibility?language=objc#2974973), meaning the application can request the secret but access is only granted if the user that stored it is physically present. This pattern is good for users but currently only possible in extensions by communicating with a native application over [Native Messaging](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Native_messaging). This is because access to both hardware based key stores and biometrics are not exposed by existing web APIs.
```
Comment (zombie) [inline proposals/secure-storage.md:13]: Perhaps a comparison between Apple's and Windows' API surfaces could be useful (later in the proposal).

File: proposals/secure-storage.md (hunk 3 above)
Comment (oliverdunk) [reply proposals/secure-storage.md:121]: @bradcush reached out out-of-band and shared this quote which he was happy for us to use in the proposal. I'm fairly sure he's happy with it being included here but I'll ask if he can drop an approval so we have everything documented in one place.

File: proposals/secure-storage.md (hunk 4 above)
Comment (oliverdunk) [reply proposals/secure-storage.md:13]: That sounds reasonable! I'll resolve this for now but open an issue once we have this merged.

File: proposals/secure-storage.md (hunk 3 above)
Comment (bradcush) [reply proposals/secure-storage.md:121]: This quote is good for us, confirming here publicly on behalf of Dashlane, thanks!

**Pull request merged**